src/groth16/prover.rs

```diff
@@ -159,6 +172,26 @@ impl<E: Engine> ConstraintSystem<E> for ProvingAssignment<E> {
     fn get_root(&mut self) -> &mut Self::Root {
         self
     }
+
+    fn is_extensible() -> bool {
+        true
+    }
+
+    fn extend(&mut self, other: &mut Self) {
```

Comment:
intuitively this should consume `other`, eg `fn extend(&mut self, other: Self)`

Reply:
I thought so too, but after a lot of pain, it seems that `BitVec` requires a mutable reference in its argument to extend, and we end up here. If you know how to make it nicer and can get it to compile, I welcome the improvement.